### PR TITLE
identify a different type of SSH URL

### LINKED ISSUE #2584: Repo is listed under the "Other" category when the protocol is SSH
### Description

When you clone a GitHub repo via SSH in the GitHub Desktop app, it is usually wrongly categorized under the "Other" repos, although it should be under the "GitHub" repos.

As a workaround, I clone my affected repos in CLI, then add them to the GitHub Desktop app.

### Version

**GitHub Desktop version:** 0.8

**OS version:** 10.12.6

### Steps to Reproduce

Clone a GitHub repository via SSH in the GitHub Desktop app. The repo I used: `git@github.com:woohoolabs/yang.git`

**Expected behavior:** The repo should be listed under the "GitHub" category

**Actual behavior:** Instead, it is listed under the "Other" category

**Reproduces how often:** Usually. I have a repo which can be cloned in GitHub Desktop well: `git@github.com:woohoolabs/yin.git`

#### Additional Information

If I can remember well, GitHub for Mac had a functionality which showed all your repos and you could clone them easily (do I remember well?). It would be awesome to have this again and be able to choose the protocol!

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #2584

This covers people who rewrite their `~/.gitconfig` to force the use of SSH:

```
[url "ssh://git@github.com/"]
    insteadOf = https://github.com/
```

Repro steps:

 - remove any `url` override entries from `~/.gitconfig`
 - clone a HTTPS GitHub repository down to my local machine using Git CLI
 - add the above entry to `~/.gitconfig`
 - drag-and-drop the Git repository onto the app to add it

### CODE REVIEW

**Comment by kocsismate** [comment]
Thank you! 😊

File: app/src/lib/remote-parsing.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export function parseRemote(url: string): IGitRemoteURL | null {
     new RegExp('^https?://(?:.+@)?(.+)/(.+)/(.+?)(?:.git/?)?$'),
     new RegExp('^git@(.+):(.+)/(.+?)(?:.git)?$'),
     new RegExp('^git:(.+)/(.+)/(.+?)(?:.git)?$'),
+    new RegExp('^ssh://git@(.+)/(.+)/(.+?)(?:.git)?$'),
```
Comment (joshaber) [inline app/src/lib/remote-parsing.ts]: Should we do the same fix as https://github.com/desktop/desktop/pull/2610 too? Probably for all the patterns, just to be safe.

Comment (shiftkey) [reply app/src/lib/remote-parsing.ts]: Yeah, good shout! ✍️

**Comment by shiftkey** [comment]
:moon:

**Pull request merged**